src/test/coqLsp/coqLspCommandInGoalsReq.test.ts

```diff
@@ -0,0 +1,239 @@
+import { expect } from "earl";
+import { Result } from "ts-results";
+
+import { createTestCoqLspClient } from "../../coqLsp/coqLspBuilders";
+import { Goal, PpString } from "../../coqLsp/coqLspTypes";
+
+import { Uri } from "../../utils/uri";
+import { resolveResourcesDir } from "../commonTestFunctions/pathsResolver";
+
+suite("Request goals with `command/pretac` argument", () => {
+    async function getGoalAtPosition(
+        position: { line: number; character: number },
+        resourcePath: string[],
+        command: string,
+        projectRootPath?: string[]
+    ): Promise<Result<Goal<PpString>[], Error>> {
+        const [filePath, rootDir] = resolveResourcesDir(
+            resourcePath,
+            projectRootPath
+        );
+        const fileUri = Uri.fromPath(filePath);
+
+        const client = await createTestCoqLspClient(rootDir);
+        await client.openTextDocument(fileUri);
+        const goals = await client.getGoalsAtPoint(
+            position,
+            fileUri,
+            1,
+            command
+        );
+        await client.closeTextDocument(fileUri);
+        client.dispose();
+
+        return goals;
+    }
+
+    test("One Coq sentence: fail to solve with bad goal", async () => {
+        const goal = await getGoalAtPosition(
+            { line: 9, character: 4 },
+            ["small_document.v"],
+            "reflexivity."
+        );
+
+        expect(goal.err).toEqual(true);
```

Comment:
The code that checks the goals extraction returned error should be moved in a separate method, so far there is pretty a lot of code duplication